lib/LightPanels/LightPanel.js

```diff
@@ -0,0 +1,200 @@
+/**
+* Contains the declaration for the {@link module:moonstone/LightPanel~LightPanel} kind.
+* @module moonstone/LightPanel
+*/
+
+var
+	kind = require('enyo/kind'),
+	LightPanel = require('enyo/LightPanels').Panel;
+
+var
+	Spotlight = require('spotlight');
+
+var
+	Header = require('../Header');
+
+/**
+* A light-weight panels implementation that has basic support for side-to-side transitions
+* between child components.
+*
+* @class LightPanel
+* @extends module:enyo/Control~Control
+* @ui
+* @public
+*/
+module.exports = kind(
+	/** @lends module:moonstone/LightPanel~LightPanel.prototype */ {
+
+	/**
+	* @private
+	*/
+	name: 'moon.LightPanel',
+
+	/**
+	* @private
+	*/
+	kind: LightPanel,
+
+	/**
+	* @private
+	*/
+	classes: 'moon-light-panel',
+
+	/**
+	* @private
+	*/
+	spotlight: 'container',
+
+		/**
+	* The title of this panel.
+	*
+	* @type {String}
+	* @default ''
+	* @public
+	*/
+	title: '',
+
+	/**
+	* Components that are part of the header.
+	*
+	* @type {Object}
+	* @default {}
+	* @public
+	*/
+	headerComponents: {},
+
+	/**
+	* The amount of time, in milliseconds, to run the fade-in animation for the panel body.
+	*
+	* @type {Number}
+	* @default 500
+	* @public
+	*/
+	clientDuration: 500,
+
+	/**
+	* The timing function to be applied to the fade-in animation for the panel body.
+	*
+	* @type {String}
+	* @default 'ease-out'
+	* @public
+	*/
+	clientTimingFunction: 'ease-out',
+
+	/**
+	* Facade for the [titleUpperCase]{@link module:moonstone/Header~Header#titleUpperCase} property
+	* of the embedded {@link module:moonstone/Header~Header}.
+	*
+	* @type {Boolean}
+	* @default true
+	* @public
+	*/
+	titleUpperCase: true,
+
+	/**
+	* @private
+	*/
+	components: [
+		{kind: Header, name: 'header', type: 'medium'},
+		{name: 'client', classes: 'client'},
+		{name: 'spotlightPlaceholder', spotlight: false, style: 'width:0;height:0;'}
+	],
+
+	/**
+	* @private
+	*/
+	bindings: [
+		{from: 'title', to: '$.header.title'},
+		{from: 'titleUpperCase', to: '$.header.titleUpperCase'}
+	],
+
+	/**
+	* @method
+	* @private
+	*/
+	create: function () {
+		LightPanel.prototype.create.apply(this, arguments);
+
+		var transValue = 'opacity ' + this.clientDuration + 'ms ' + this.clientTimingFunction;
+		this.$.client.applyStyle('-webkit-transition', transValue);
+		this.$.client.applyStyle('transition', transValue);
+
+		if (this.headerComponents) {
+			var owner = this.hasOwnProperty('headerComponents') ? this.getInstanceOwner() : this;
+			this.$.header.createComponents(this.headerComponents, {owner: owner});
+		}
+	},
+
+	/**
+	* This overridable method is called before a transition.
+	*
+	* @public
+	*/
+	preTransition: function () {
```

Comment:
Yeah thought about that but opted not to because of the no-op. I agree it does set us up better for future work, and is self-documenting in that sense.